galaxy_ng/app/auth/auth.py

```diff
@@ -44,13 +44,28 @@ def authenticate(self, request):
 
         header = self._decode_header(request.META[self.header])
 
-        try:
-            identity = header['identity']
-            account = identity['account_number']
-
-            user = identity['user']
-            username = user['username']
-        except KeyError:
+        if header.get("identity", {}).get("type") == "User":
+            try:
+                identity = header['identity']
+                account = identity['account_number']
+
+                user = identity['user']
+                username = user['username']
+            except KeyError:
+                raise AuthenticationFailed
+        elif header.get("identity").get("type") == "ServiceAccount":
```

Comment:
do you want the `, {}` in the first `get`?

```suggestion
        elif header.get("identity", {}).get("type") == "ServiceAccount":
```